clients/src/main/java/org/apache/kafka/common/cache/Cache.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package org.apache.kafka.common.cache;
+
+/**
+ * Interface for caches, semi-peristent maps which store key-value mappings until either an eviction criteria is met
+ * or the entries are manually invalidated. Caches are not required to be thread-safe, but some implementations may be.
+ */
+public interface Cache<K, V> {
```

Comment:
Agreed that this falls under the same category as the utils discussion.

If we decide that we want to provide commonly-used classes like this one, one option would be to state that this interface should not be implemented by clients. This is an approach that is used often in the Eclipse Platform (they have extensive rules on compatibility rules: https://wiki.eclipse.org/Evolving_Java-based_APIs_2).

If we do that, we would have the freedom to add new methods if and when needed. I don't foresee the need to change the signature of existing methods or remove methods since they're pretty standard and there are only 3 of them.